distribution/std-lib/Database/src/Connection/Connection.enso

```diff
@@ -23,7 +35,174 @@ type Connection
 
        Arguments:
        - name: name of the table to access
-    access_table : Text -> Table
-    access_table name =
-        column_names = this.internal_connection.fetch_column_names name
-        Table.make_table this name column_names
+    access_table : Text -> Database_Table
+    access_table name = here.wrap_sql_errors <|
+        columns = this.fetch_columns name
+        Database_Table.make_table this name columns
+
+    ## Closes the connection releasing the underlying database resources
+       immediately instead of waiting for them to be automatically released.
+
+       The connection is not usable afterwards.
+    close : Nothing
+    close =
+        Managed_Resource.finalize this.connection_resource
+
+    ## ADVANCED
+
+       Executes a raw query and returns the result as an in-memory Table.
+
+       Arguments:
+       - query: either raw SQL code as Text or an instance of Sql.Statement
+         representing the query to execute.
+       - expected_types: an optional array of expected types of each column;
+         meant only for internal use.
+    execute_query : Text | Sql.Statement -> Vector Sql.Sql_Type -> Materialized_Table =
+    execute_query query expected_types=Nothing = here.wrap_sql_errors <|
```

Comment:
I'm thinking whether this one and `execute_update` should be marked as unsafe, since they allow to execute arbitrary SQL which can be dangerous.